### PR TITLE
Support huggingface chat template

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

**Comment by AllentDan** [comment]
This PR needs lots of tests and comments.

File: lmdeploy/api.py [hunk 1]
```diff
@@ -76,9 +76,10 @@ def serve(model_path: str,
           chat_template_config: Optional[ChatTemplateConfig] = None,
           server_name: str = '0.0.0.0',
           server_port: int = 23333,
-          log_level: str = 'ERROR',
+          log_level: str = 'WARNING',
           api_keys: Optional[Union[List[str], str]] = None,
           ssl: bool = False,
+          block: bool = False,
```
Comment (lvhan028) [inline lmdeploy/api.py:82]: Is it related to chat template?

Comment (AllentDan) [reply lmdeploy/api.py:82]: No. Should I remove it in this PR? I can renew a PR for it.

Comment (lvhan028) [reply lmdeploy/api.py:82]: I think it's better to pull another PR.

File: src/turbomind/models/llama/LlamaBatch.cc [hunk 2]
```diff
@@ -235,7 +235,7 @@ void LlamaBatch<T>::ProcessInferRequests(const Requests& requests)
         FT_CHECK(!state.requests[idx]);
 
         if (rank_ == 0) {
-            TM_LOG_INFO("[ProcessInferRequests] Request for %ld received.", (long)r->id);
+            TM_LOG_WARNING("[ProcessInferRequests] Request for %ld received.", (long)r->id);
```
Comment (lvhan028) [inline src/turbomind/models/llama/LlamaBatch.cc:238]: Why changing it from INFO to WARNING?

File: docs/en/inference/pipeline.md [hunk 3]
```diff
@@ -235,6 +235,66 @@ This class contains the generation parameters used by inference engines.
 | min_new_tokens      | int         | The minimum numbers of tokens to generate, ignoring the number of tokens in the prompt.                               | None    |
 | skip_special_tokens | bool        | Whether or not to remove special tokens in the decoding.                                                              | True    |
 
+## Customize chat template
```
Comment (lvhan028) [inline docs/en/inference/pipeline.md:238]: 自定义对话模板，可以单独写到一个文档中。pipeline.md，restful_api.md 可以索引它

**Comment by fzyzcjy** [comment]
Hi, is there any updates? Thanks!

**Pull request closed without merging**